### ISSUE TITLE
取方块与放方块的一个的想法

### ISSUE THREAD

**Issue by zyalll** [MICF-JI204/SocMain#38]
![u11 o ojb yjb mk ps1](https://cloud.githubusercontent.com/assets/11556422/6768101/b7119350-d092-11e4-94f3-b044f1c47156.png)

图太渣。。
但是还是如图。。
我的想法主要是两个部分。 一部分是储存木块区，另一部分是横扫进攻区。
1，储存木块区。
这里借鉴了LC大神的想法。
首先因为如果把木块储存放在外面会造成重心不稳 ，所以我想把这个部分放在车内部，这样木块放多了只会对车重造成影响而不至于翻车。
我们先用机械手抓起木块，举起来后机械手转个方向把木块丢到存放区内。因为有了存放区这个设定，我们不需要担心机械手一次只能搞一个而效率过低。并且无论使用逐个搬运或者近乎整体搬运的策略都可以实现。（当然，这是一种假【点】整体搬运，实际效率肯定要低于真【点】整体搬运，但是对木块操作的精度有所提升）
2，关于进攻区
因为CJ大神提出的模型里车是可以原地旋转的，因此我想到这样的方法。
如图在车尾部伸出一个坚硬的尾巴，进攻时只需要靠近圆台转啊转啊就好了，考虑方形结构会可能卡住，因此车整体为圆形为佳。（但是我TMD手残画不出来，各位脑补一下。。）
3，关于如何放置木块
在车内部有个圆圆的东西，并且进攻杆内端也是弧形的，这样在我们要放置木块时只需要将机械手那端靠近圆台，然后收回进攻杆，同时内部存放区右端升起形成一个斜面，木块哗啦啦全部出去了。并且因为在出装置时有一定的初速度，只要机械手不是非常粗壮不用担心机械手挡着木块（会不会砸坏机械手？。。。）
有这个考虑的一个原因是我们的进攻和放木块并不是同时进行的，在车检时我们把进攻杆收回，进场后把进攻杆放下并开始储存木块。

策略：开始尽量收集木块，收集一定的木块后转为单个搬运策略，去搬运竞争台上对方木块并置于己方得分区，最后半分钟横扫两个圆台，然后哗啦啦把木块倒出去。
问题：1，感觉机械手的制作会比较复杂，要既能抓到木块又可以旋转到小车内部上方把木块扔进去。
2，在搬运对方木块时只能逐个搬运，效率很低（但是如果想用对方木块在得分区堆叠以产生加成的话似乎只能用精度高的机械手。）
3，进攻杆不能太长，因此并不能覆盖整个圆台。

**Comment by wzy199671** [comment]
我能改下标题么QAQ。。。标题忘换啦。。

**Comment by Heim12** [comment]
能不能直接把机械手伸直兼任进攻的任务（是担心会损坏机械手？）

**Comment by tripack45** [comment]
应该可以吧= =机械手目测很重（因为上面有电机）所以应该砸不坏

**Comment by zyalll** [comment]
@Heim12  对~如果就用机械手进行进攻的话设计会简单不少，不需要在带上长长的尾巴了，内部设计也会更简单些。只是我不知道机械手臂是否可以进行伸直之类的操作。。
